pkg/compiler/debug.go

```diff
@@ -169,97 +176,111 @@ func (c *codegen) registerDebugVariable(name string, expr ast.Expr) {
-		Range:      scope.rng,
-		Parameters: params,
-		ReturnType: c.scReturnTypeFromScope(scope),
-		SeqPoints:  c.sequencePoints[name],
-		Variables:  scope.variables,
+		IsExported:   scope.decl.Name.IsExported(),
+		IsFunction:   scope.decl.Recv == nil,
+		Range:        scope.rng,
+		Parameters:   params,
+		ReturnType:   vt,
+		ReturnTypeSC: st,
+		SeqPoints:    c.sequencePoints[name],
+		Variables:    scope.variables,
 	}
 }
 
-func (c *codegen) scReturnTypeFromScope(scope *funcScope) string {
+func (c *codegen) scAndVMReturnTypeFromScope(scope *funcScope) (smartcontract.ParamType, string) {
 	results := scope.decl.Type.Results
 	switch results.NumFields() {
 	case 0:
-		return "Void"
+		return smartcontract.VoidType, "Void"
 	case 1:
-		return c.scTypeFromExpr(results.List[0].Type)
+		st, vt := c.scAndVMTypeFromExpr(results.List[0].Type)
+		return st, vt.String()
 	default:
 		// multiple return values are not supported in debugger
-		return "Any"
+		return smartcontract.AnyType, "Any"
 	}
 }
 
-func (c *codegen) scTypeFromExpr(typ ast.Expr) string {
+func (c *codegen) scAndVMTypeFromExpr(typ ast.Expr) (smartcontract.ParamType, stackitem.Type) {
 	t := c.typeOf(typ)
 	if c.typeOf(typ) == nil {
-		return "Any"
+		return smartcontract.AnyType, stackitem.AnyT
 	}
 	if named, ok := t.(*types.Named); ok {
 		if isInteropPath(named.String()) {
 			name := named.Obj().Name()
 			pkg := named.Obj().Pkg().Name()
 			switch pkg {
 			case "blockchain", "contract":
-				return "Array" // Block, Transaction, Contract
+				return smartcontract.ArrayType, stackitem.ArrayT // Block, Transaction, Contract
 			case "interop":
 				if name != "Interface" {
-					return name
+					switch name {
+					case "Hash160":
+						return smartcontract.Hash160Type, stackitem.ByteArrayT
+					case "Hash256":
+						return smartcontract.Hash256Type, stackitem.ByteArrayT
+					case "PublicKey":
+						return smartcontract.PublicKeyType, stackitem.ByteArrayT
+					case "Signature":
+						return smartcontract.SignatureType, stackitem.ByteArrayT
+					}
 				}
 			}
-			return "InteropInterface"
+			return smartcontract.InteropInterfaceType, stackitem.InteropT
 		}
 	}
 	switch t := t.Underlying().(type) {
 	case *types.Basic:
 		info := t.Info()
 		switch {
 		case info&types.IsInteger != 0:
-			return "Integer"
+			return smartcontract.IntegerType, stackitem.IntegerT
 		case info&types.IsBoolean != 0:
-			return "Boolean"
+			return smartcontract.BoolType, stackitem.BooleanT
 		case info&types.IsString != 0:
-			return "String"
+			return smartcontract.StringType, stackitem.ByteArrayT
 		default:
-			return "Any"
+			return smartcontract.AnyType, stackitem.AnyT
 		}
 	case *types.Map:
-		return "Map"
+		return smartcontract.MapType, stackitem.MapT
 	case *types.Struct:
-		return "Struct"
+		return smartcontract.ArrayType, stackitem.StructT
 	case *types.Slice:
 		if isByte(t.Elem()) {
-			return "ByteString"
+			return smartcontract.ByteArrayType, stackitem.BufferT
```

Comment:
It might as well be `stackitem.ByteArrayT`.

Reply:
Fixed.